src/services/push-notifications-api/index.ts

```diff
@@ -0,0 +1,52 @@
+import PushNotification from 'react-native-push-notification';
+import PushNotificationIOS from '@react-native-community/push-notification-ios';
+import pushHandlers from 'src/push-handlers';
+
+type PushNotificationArgs = {
+  title: string;
+  body: string;
+  type: string;
+  payload?: any;
+};
+
+class PushNotificationsApi {
+  constructor() {
+    PushNotification.configure({
+      onNotification: notification => {
+        const handlersToCall = pushHandlers.filter(handler =>
+          handler.types.includes(notification.data.type)
+        );
+
+        handlersToCall.forEach(handler =>
+          handler.action(notification.data.type, notification.data.payload)
+        );
+
+        notification.finish(PushNotificationIOS.FetchResult.NoData);
```

Comment:
Why IOS here?

Reply:
Because this parameter is for IOS. It makes no sense and is ignored on Android.